### ISSUE TITLE
DirectoryIndex won't work in apache 2.4

### ISSUE THREAD

**Issue by lonoak** [UNINETT/mod_auth_mellon#38]
Hi, 
  as I have pointed in the ML, there's an issue which happens when the access URL does not end with a file but a directory instead; a directory which contains a valid  index file defined with Apache's DirectoryIndex directive. This seems to work under Apache 2.2 but for some reason, no longer in Apache 2.4.

Here's the mailing list topic, just for reference:

https://postlister.uninett.no/sympa/arc/modmellon/2015-07/msg00002.html

Jose.

**Comment by thijskh** [comment]
I can confirm this behaviour with the most recent release of mod_auth_mellon, 0.10.0 with Apache 2.4.

It looks like Apache performs several subrequests to try each of the DirectoryIndex options which all fail (while this directory does have an `index.html`, which is accessible when requested explicitly):
```
[Mon Jul 13 14:29:50.566275 2015] [core:error] [pid 16598] [client 195.169.126.239:44278] AH00027: No authentication done but request not allowed without authentication for /testmeloen/testdir/index.html. Authentication not configured?
[Mon Jul 13 14:29:50.566785 2015] [core:error] [pid 16598] [client 195.169.126.239:44278] AH00027: No authentication done but request not allowed without authentication for /testmeloen/testdir/index.cgi. Authentication not configured?
[Mon Jul 13 14:29:50.566899 2015] [core:error] [pid 16598] [client 195.169.126.239:44278] AH00027: No authentication done but request not allowed without authentication for /testmeloen/testdir/index.pl. Authentication not configured?
[Mon Jul 13 14:29:50.567016 2015] [core:error] [pid 16598] [client 195.169.126.239:44278] AH00027: No authentication done but request not allowed without authentication for /testmeloen/testdir/index.php. Authentication not configured?
[Mon Jul 13 14:29:50.567136 2015] [core:error] [pid 16598] [client 195.169.126.239:44278] AH00027: No authentication done but request not allowed without authentication for /testmeloen/testdir/index.xhtml. Authentication not configured?
[Mon Jul 13 14:29:50.567252 2015] [core:error] [pid 16598] [client 195.169.126.239:44278] AH00027: No authentication done but request not allowed without authentication for /testmeloen/testdir/index.htm. Authentication not configured?
```

**Comment by olavmrk** [comment]
I'm tempted to just drop the entire block of code that attempts to bypass authentication for subrequests. The code predates my involvement on the project, but If I understand it correctly, it is basically an optimization, which now has turned out to not really work as it should.

**Comment by olavmrk** [comment]
I think this bug is fixed in 2ac9dce, which will be included in version 0.11.0.
